src/card/interface.js

```diff
@@ -65,17 +67,19 @@ export function getCardFields() : ?Card {
         return cardFrame.getFieldValue();
     }
 
-    const { cardNumberFrame, cardCVVFrame, cardExpiryFrame } = getCardFrames();
+    const { cardNumberFrame, cardCVVFrame, cardExpiryFrame, cardNameFrame } = getCardFrames();
 
     if (
         cardNumberFrame && cardNumberFrame.isFieldValid() &&
         cardCVVFrame && cardCVVFrame.isFieldValid() &&
-        cardExpiryFrame && cardExpiryFrame.isFieldValid()
+        cardExpiryFrame && cardExpiryFrame.isFieldValid() &&
+        (cardNameFrame ? cardNameFrame.isFieldValid() : true)
     ) {
         return {
             number: cardNumberFrame.getFieldValue(),
             cvv:    cardCVVFrame.getFieldValue(),
-            expiry: cardExpiryFrame.getFieldValue()
+            expiry: cardExpiryFrame.getFieldValue(),
+            name:   cardNameFrame && cardNameFrame.isFieldValid() ? cardNameFrame.getFieldValue() : ''
```

Comment:
can you explain this extra logic here?

Reply:
@crookedneighbor they way SPB gets the value for each field from the iFrames is by getting the frames which has access to the `win.exports` (zoid). 

Here is the function use for this purpose ---> https://github.com/paypal/paypal-smart-payment-buttons/blob/master/src/card/interface.js#L16

Having access to the frame then we have access for the methods defined on [Page](https://github.com/paypal/paypal-smart-payment-buttons/blob/master/src/card/components/page.jsx) for each field.

For the particular case of the `name` field which is optional I'm adding a logic to first check if the frame was defined (meaning that the field was set by the user), checking validity and if true returning the field value.

Reply:
Would this syntax work in flow? If the object exists, call `getFieldValue()` and default any falsy values to `""`.

```suggestion
            name:   cardNameFrame?.getFieldValue() || ''
```

Reply:
@crookedneighbor It worked, I'm pushing a change.